aep/0122.yaml

```diff
@@ -0,0 +1,150 @@
+aliases:
+  AepResource:
+    description: A schema that represents an AEP resource
+    targets:
+      - formats: ['oas2', 'oas3']
+        given:
+          - $.components.schemas[?(@.x-aep-resource)]
+          - $.definitions[?(@.x-aep-resource)]
+
+rules:
+  aep-122-resource-path-field:
+    description: Resources must have a 'path' field of type string.
+    message: Resource schema must include a 'path' property of type 'string'
+    severity: error
+    formats: ['oas2', 'oas3']
+    given:
+      - $.components.schemas[?(@['x-aep-resource'])]
+      - $.definitions[?(@['x-aep-resource'])]
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          type: object
+          properties:
+            properties:
+              type: object
+              properties:
+                path:
+                  type: object
+                  properties:
+                    type:
+                      const: string
+                  required: ['type']
+              required: ['path']
+          required: ['properties']
+
+  aep-122-path-field-required:
+    description: The 'path' field must be in the required array.
+    message: The 'path' field must be listed in the 'required' array
+    severity: error
+    formats: ['oas2', 'oas3']
+    given:
+      - $.components.schemas[?(@['x-aep-resource'])]
+      - $.definitions[?(@['x-aep-resource'])]
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          type: object
+          properties:
+            required:
+              type: array
+              contains:
+                const: path
+          required: ['required']
```

Comment:
It's not completely clear in AEP-122 if the `path` field is required. In particular, the OAS example _does not_ show it as required. I've asked for clarification on this in Slack.